harness/determined/experimental/inference/_torch_batch_process.py

```diff
@@ -0,0 +1,412 @@
+import abc
+import json
+import logging
+import math
+import os
+import pathlib
+import uuid
+import warnings
+from typing import Any, Dict, Iterator, Optional, Set, Type
+
+import torch
+import torch.distributed as dist
+from torch import nn
+from torch.utils.data import Dataset
+
+import determined as det
+from determined import core, pytorch
+from determined.common import set_logger
+from determined.pytorch import DataLoader
+
+set_logger(False)
+
+DEFAULT_BATCH_SIZE = 1
+
+
+class TorchBatchProcessorContext(pytorch._PyTorchReducerContext):
+    def __init__(self, core_context: core.Context, storage_path: str):
+        super().__init__()
+        self._core_context = core_context
+        self._device = get_default_device(core_context)
+        self._tensorboard_path = core_context.train.get_tensorboard_path()
+        self._storage_path = storage_path
+        self._use_default_storage = False
+
+    def to_device(self, data, warned_types: Optional[Set[Type]] = None) -> pytorch.TorchData:
+        """Map generated data to the device allocated by the Determined cluster.
+
+        All the data in the data loader and the models are automatically moved to the
+        allocated device. This method aims at providing a function for the data generated
+        on the fly.
+        """
+        return pytorch.to_device(data, self._device, warned_types)
+
+    def get_tensorboard_path(self) -> pathlib.Path:
+        """
+        Tensorboard files should be written to the path returned to be shown properly in the UI.
+        For example, the path should be passed to PyTorch profiler as shown below:
+
+        torch.profiler.profile(
+            activities=...,
+            schedule=...,
+            on_trace_ready=torch.profiler.tensorboard_trace_handler(<tensorboard_path>),
+        )
+        """
+        return self._tensorboard_path
+
+    def get_device(self):
+        """
+        Get default device associated with this worker
+        """
+        return self._device
```

Comment:
I would _slightly_ prefer just `context.device` to match the PyTorchTrialContext.  Just slightly though.